src/main/java/io/dockstore/provision/GSPlugin.java

```diff
@@ -0,0 +1,300 @@
+import com.google.cloud.WriteChannel;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageException;
+import com.google.cloud.storage.StorageOptions;
+import com.google.common.collect.Lists;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.apache.commons.lang3.StringUtils;
+import ro.fortsoft.pf4j.Extension;
+import ro.fortsoft.pf4j.Plugin;
+import ro.fortsoft.pf4j.PluginWrapper;
+import ro.fortsoft.pf4j.RuntimeMode;
+
+// Imports the Google Cloud client library
+
+/**
+ * @author wshands
+ */
+public class GSPlugin extends Plugin {
+
+    public GSPlugin(PluginWrapper wrapper) {
+        super(wrapper);
+    }
+
+    @Override
+    public void start() {
+        // for testing the development mode
+        if (RuntimeMode.DEVELOPMENT.equals(wrapper.getRuntimeMode())) {
+            System.out.println(StringUtils.upperCase("GSPlugin development mode"));
+        }
+    }
+
+    @Override
+    public void stop() {
+        System.out.println("GSPlugin.stop()");
+    }
+
+    @Extension
+    public static class GSProvision implements ProvisionInterface {
+
+        private static final int MAX_FILE_SIZE_FOR_SINGLE_WRITE = 1_000_000;
+        private static final int DOWNLOAD_BUFFER_SIZE = 64 * 1024;
+        private static final int UPLOAD_BUFFER_SIZE = 1024;
+        private static final String DEFAULT_CONTENT_TYPE = "application/octet-stream";
+        private Map<String, String> config;
+
+        public void setConfiguration(Map<String, String> map) {
+            this.config = map;
+        }
+
+        private Storage getGoogleGSClient() {
+            return StorageOptions.getDefaultInstance().getService();
+        }
+
+        private List<String> getSplitPathList(String path) {
+            String trimmedPath = path.replace("gs://", "");
+            return Lists.newArrayList(trimmedPath.split("/"));
+        }
+
+        protected String getBucketName(String path) {
+            List<String> splitPathList = getSplitPathList(path);
+            return splitPathList.remove(0);
+        }
+
+        // Get the path to the source file minus the scheme and bucket name
+        protected String getBlobName(String path) {
+            List<String> splitPathList = getSplitPathList(path);
+            // Remove the bucket name from the path
+            List<String> splitPathListNoBucket = splitPathList.subList(1, splitPathList.size());
+            return String.join(File.separator, splitPathListNoBucket);
+        }
+
+        public Set<String> schemesHandled() {
+            return new HashSet<>(Lists.newArrayList("gs"));
+        }
+
+        public boolean downloadFrom(String sourcePath, Path destination) {
+            Storage gsClient = getGoogleGSClient();
+
+            //String trimmedPath = sourcePath.replace("gs://", "");
```

Comment:
Unless there's a good chance commented code will be uncommented (and there should be a comment why), don't commit commented code (good summary of my general train of thought https://blog.kentcdodds.com/please-don-t-commit-commented-out-code-53d0b5b26d5f )

Reply:
removed commented code